File: Ruby/ShieldBooster.rb
#encoding:utf-8

require_relative 'ShieldToUI'

module Deepspace

    class ShieldBooster

        def initialize(_name,_boost,_uses)
            @name = _name
            @boost = _boost
            @uses = _uses
        end

        def self.newCopy(sb)
            new(sb.name,sb.boost,sb.uses)
        end

        def useIt
            ret = 1.0
            if @uses > 0
                @uses = @uses - 1
                ret = @boost
            end

            ret
        end

        def getUIversion
            ShieldToUI.new(self)
        end

        def to_s()
            "This Shield Booster is:" + @name + ", with boost:" + @boost + ", and uses:" + @uses
        end

        attr_reader :boost, :uses, :name
        
    end

end

File: Java/deepspace/src/deepspace/ShieldBooster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author fjaviersaezm
 */
class ShieldBooster implements CombatElement {
    private String name;
    private float boost;
    private int uses;
    
    

    ShieldBooster(String name, float boost, int uses) {
        this.name = name;
        this.boost = boost;
        this.uses = uses;
    }
    
    ShieldBooster(ShieldBooster s) {
        this(s.name,s.boost,s.uses);
    }

    public float getBoost() {
        return boost;
    }

    @Override
    public int getUses() {
        return uses;
    }
    
    @Override
    public float useIt(){
        float ret = 1.0f;
        
        if(this.uses > 0){
            this.uses -= 1;
            ret = this.boost;
        }
        
        return ret;
    }
    
    public String toString(){
        String s = String.format("This ShieldBooster has: name = %s , boost = %f , uses = %d .", name,boost,uses);
        return s;
    }
    
    ShieldToUI getUIversion(){
        return new ShieldToUI(this);
    }
}
